lib/internal/streams/destroy.js

```diff
@@ -52,23 +60,44 @@ function destroy(err, cb) {
     r.destroyed = true;
   }
 
-  this._destroy(err || null, (err) => {
+  // If still constructing then defer calling _destroy.
+  if (!s.constructed) {
+    if (err || cb) {
+      // Slow path. Allocating closure. This should very rarely happen.
+      // This path can only be taken if the stream is destroyed with
+      // an error or a callback before it has been constructed.
+      s[kDestroy] = (stream, er) => {
+        _destroy(stream, err || er, cb);
```

Comment:
@mcollina: Which error should take priority here?

`err`: from the `.destroy(err)` call? Occurs first.
`er`: from the `_construct` callback? Occurs last.

Reply:
@lpinca

Reply:
A user calling `.destroy(err)` intends for that error to be surfaced, so that one should win.

Reply:
I would say the first one which is also how it works in `_destroy()` implementation IIRC.

Reply:
Cool, I think it's the correct the way it is then.